File: backend/models/TechnicalInterviewReport.js
const mongoose = require('mongoose');

const technicalInterviewReportSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  interviewId: {
    type: String,
    required: true,
    index: true
  },
  question: {
    type: String,
    required: true
  },
  answer: {
    type: String,
    required: true
  },
  technicalScore: Number,
  communicationScore: Number,
  grammarScore: Number,
  confidenceScore: Number,
  keywordScore: Number,
  overallScore: Number,
  feedback: String,
  strength: String,
  weakness: String,
  recommendation: String
}, { timestamps: true });

module.exports = mongoose.model(
  'TechnicalInterviewReport',
  technicalInterviewReportSchema,
  'technical_interview_reports'
);

File: backend/models/technical_interview_report.py
"""models/technical_interview_report.py — TechnicalInterviewReport table"""
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey, Text
from sqlalchemy.sql import func
from database import Base


class TechnicalInterviewReport(Base):
    __tablename__ = "technical_interview_reports"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    interview_id = Column(String(255), nullable=False, index=True)
    question = Column(Text, nullable=False)
    answer = Column(Text, nullable=False)
    technical_score = Column(Float, nullable=True)
    communication_score = Column(Float, nullable=True)
    grammar_score = Column(Float, nullable=True)
    confidence_score = Column(Float, nullable=True)
    keyword_score = Column(Float, nullable=True)
    overall_score = Column(Float, nullable=True)
    feedback = Column(Text, nullable=True)
    strength = Column(Text, nullable=True)
    weakness = Column(Text, nullable=True)
    recommendation = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
